src/model_api.jl

```diff
@@ -182,3 +184,55 @@ If for some reason a model is sometimes unable to report feature importances the
 
 """
 function feature_importances end
+
+_named_tuple(named_tuple::NamedTuple) = named_tuple
+_named_tuple(::Nothing) = NamedTuple()
+_named_tuple(something_else) = (report=something_else,)
+_scrub(x) = x
+_scrub(x::NamedTuple) = isempty(x) ? nothing : x
+_keys(named_tuple) = keys(named_tuple)
+_keys(::Nothing) = ()
+
+"""
+    MLJModelInterface.report(model, report_given_method)
+
+Merge the reports in the dictionary `report_given_method` into a single
+property-accessible object. It is supposed that the keys of the dictionary are
+`:fit` and the symbolic names of MLJModelInterface.jl operations, such as `:predict` or
+`:transform`. Each value will be the `report` component returned by a training method
+(`fit` or `update`) dispatched on the `model` type, in the case of `:fit`, or the
+report component returned by an operation that supports reporting.
+
+# New model implementations
+
+Overloading this method is optional, unless the model generates reports that are neither
+named tuples nor `nothing`.
+
+Assuming each dictionary value is a named tuple or `nothing`, the fallback returns the
+usual named tuple merge of the dictionary values, ignoring any `nothing` values, and
+assuming there are no conflicts between the keys of the dictionary values (the individual
+reports). If there is a key conflict, all operation reports are first wrapped in a named
```

Comment:
At first this seemed like we have multiple dictionaries.

```suggestion
Assuming each value in the `report_given_method` dictionary is either a named tuple 
or `nothing`, and there are no conflicts between the keys of the dictionary values 
(the individual reports), the fallback returns the usual named tuple merge of the 
dictionary values, ignoring any `nothing` value. If there is a key conflict, all operation 
reports are first wrapped in a named
```

Reply:
Yes, much better! Thanks.